boards/arm/nucleo_u5a5zj_q/nucleo_u5a5zj_q.dts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include "nucleo_u5a5zj_q-common.dtsi"
+
+/ {
+	model = "STMicroelectronics STM32U5A5ZJ-NUCLEO-Q board";
+	compatible = "st,stm32u5a5zj-nucleo-q";
+
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	chosen {
+		zephyr,console = &usart1;
+		zephyr,shell-uart = &usart1;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,canbus = &fdcan1;
+	};
+
+	aliases {
+		led0 = &blue_led_1;
+		sw0 = &user_button;
+		pwm-led0 = &pwm_led_1;
+		pwm-led1 = &pwm_led_2;
+		watchdog0 = &iwdg;
+		volt-sensor0 = &vref1;
+		volt-sensor1 = &vbat4;
+	};
+};
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/*
+		 * Following flash partition is dedicated to the use of nucleo_u5a5zj_q
+		 * with TZEN=0 (so w/o TFM).
+		 * Set the partitions with first MB to make use of the whole Bank1
+		 */
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 DT_SIZE_K(64)>;
+		};
+		slot0_partition: partition@10000 {
+			label = "image-0";
+			reg = <0x00010000 DT_SIZE_K(1952)>;
+		};
+		slot1_partition: partition@78000 {
+			label = "image-1";
+			reg = <0x00078000 DT_SIZE_K(1952)>;
+		};
+		scratch_partition: partition@e0000 {
+			label = "image-scratch";
+			reg = <0x000e0000 DT_SIZE_K(64)>;
+		};
```

Comment:
Swap using scratch is not recommended for use, would just increase slot1 by 1 sector then use the rest of this for something else e.g. storage and set mcuboot to swap using move mode (then test updating with an application)

Reply:
It is done the same way in many other STM32 boards (like nucleo_h743zi or b_u585i_iot02a).
If this method is not recommended, should they all be updated eventually?

Reply:
There are problems with swap using scratch that have never been fully identified or understood, the only places it is ideal is where you have giant sectors (e.g. stm32f4 with 128KB sectors) or if the sector sizes of the primary and secondary slots differ (i.e. external QSPI flash having larger/smaller sector sizes than internal flash)

Reply:
Ok, understood. I have updated the partitions